components/sup/src/manager/file_watcher.rs

```diff
@@ -2685,7 +2687,11 @@ mod tests {
                                       });
             if self.parent_is_watched(&pp) {
                 // One event - create.
-                1
+                if !self.is_poll_watcher() {
+                    1
+                } else {
+                    5
+                }
```

Comment:
Here and all the other places where you are returning the number of events feels hacky to handle here because the interface is now misleasing. Seems like you are trying to return an arbitrary number of events for the pollwatcher which does not actually match up with any actual event count. Looks like this is all in service of bumping the iteration count later. Seems like things would be more clear and involve less code changes if you consolidated this in `spin_watcher` and have that multiply the iterations for the poll watcher.